crates/pathfinder/src/ethereum/log.rs

```diff
@@ -73,27 +73,83 @@ async fn get_logs<T: Transport>(
     transport: &Web3<T>,
     filter: Filter,
 ) -> Result<Vec<web3::types::Log>, GetLogsError> {
+    use web3::Error::*;
+    use RpcErrorCode::*;
     /// Error message generated by spurious decoder error which occurs on Infura endpoints from
     /// time to time. It appears that the returned value is simply empty.
     const DECODER_ERR: &str =
         "Error(\"invalid type: null, expected a sequence\", line: 0, column: 0)";
     loop {
         match transport.eth().logs(filter.clone()).await {
             Ok(logs) => return Ok(logs),
-            Err(web3::Error::Rpc(err)) if err.code.code() == RpcErrorCode::LimitExceeded.code() => {
-                return Err(GetLogsError::QueryLimit)
+            Err(Rpc(err)) if err.code.code() == LimitExceeded.code() => {
+                return Err(GetLogsError::QueryLimit);
```

Comment:
Maybe it would be useful to `trace!` when we hit the limit on infura & alchemy too?

Reply:
I wasn't sure since its a valid, expected response. And it does occur quite regularly. If anything I guess one would add these traces in the calling function?

Reply:
Oh,if it happens often then yes, definitely.